### PR TITLE
Update s390x actions-runner docker

### PR DESCRIPTION
**Pull request by Dead2**
- Updated to new actions runner version, requiring .Net 8
- Update and simplify README
- Add script for auto(re)building, suitable both for weekly cronjob and manual builds
- Improved readability, error handling and log debug output
- Moved all files to a single folder
- Removed locally hosted patches, instead fetching from gaplib

## Summary by CodeRabbit

- **New Features**
  - Updated instructions for configuring DFLTCC hardware acceleration support.
  - Introduced a new Bash script for starting an ephemeral GitHub Actions runner.
  - Added a shell script for managing the lifecycle of a self-hosted GitHub Actions runner.
  - Updated the Dockerfile to enhance package installation and patch application methods.

- **Bug Fixes**
  - Enhanced stopping process for the GitHub Actions runner service.

- **Documentation**
  - Restructured and clarified README instructions for setup and management processes.

- **Chores**
  - Removed outdated scripts and configuration files related to the runner setup.

### CODE REVIEW

**Review by iii-i — APPROVED** [review]
LGTM. I will bug the gaplib maintainers about the .NET 8 support.

File: arch/s390/self-hosted-builder/actions-runner.Dockerfile [hunk 1]
```diff
@@ -12,15 +12,13 @@ RUN     dnf install -y -q dotnet-sdk-6.0 && \
...
 
 RUN     cd /tmp && \
         git clone -q https://github.com/actions/runner && \
         cd runner && \
-        git checkout $(git describe --tags $(git rev-list --tags --max-count=1)) -b build && \
+        git checkout v2.317.0 -b build && \
         git apply /tmp/runner.patch && \
-        cp -f /tmp/global.json src/global.json
-
+       sed -i'' -e /version/s/6......\"$/${SDK}.0.100\"/ src/global.json
```
Comment (mtl1979) [inline arch/s390/self-hosted-builder/actions-runner.Dockerfile]: Indentation seems off here...

Comment (Dead2) [reply arch/s390/self-hosted-builder/actions-runner.Dockerfile]: Fixed

**Comment by nmoinvaz** [comment]
@iii-i are these the issues https://github.com/anup-kodlekere/gaplib/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+author%3Aiii-i?

**Comment by iii-i** [comment]
Yes. It's unfortunate that it's now broken for that long; the gaplib maintainer told me that the latest runner version was updated to use dotnet6 instead of dotnet8, and this made the fix complicated, so he wanted to wait until they bump it back to dotnet8, which happened today: https://github.com/actions/runner/pull/3500. So hopefully there will be some progress now.

**Comment by iii-i** [comment]
With the latest set of changes, I'm finally able to build gaplib again (specifically, this commit: https://github.com/anup-kodlekere/gaplib/tree/f5085ab2e51fad20329b441f3c5475c50e4ea1bd). Could you please give it a try?

**Review by nmoinvaz — APPROVED** [review]

File: arch/s390/self-hosted-builder/actions-runner-rebuild.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+podman image rm localhost/zlib-ng/actions-runner
+
+# Build image
+podman build --squash -f actions-runner.Dockerfile --tag zlib-ng/actions-runner --build-arg .
+
+# Create container
+podman create --replace --name=gaplib-actions-runner --env-file=/etc/actions-runner --init --volume=actions-runner-temp:/home/actions-runner zlib-ng/actions-runner
+
+# Start service
+systemctl start actions-runner
```
[the remark this answers is not in the document]
Comment (Dead2) [reply arch/s390/self-hosted-builder/actions-runner-rebuild.sh]: Added some error handling and validation.

File: arch/s390/self-hosted-builder/actions-runner [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Ephemeral runner startup script.
+#
+# Expects the following environment variables:
+#
+# - REPO=<owner>
+# - PAT_TOKEN=<github_pat_***>
+#
+
+set -e -u
```
[the remark this answers is not in the document]
Comment (Dead2) [reply arch/s390/self-hosted-builder/actions-runner:12]: Validation added.

File: arch/s390/self-hosted-builder/actions-runner [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    --url "https://github.com/$REPO" \
+    --token "$REG_TOKEN" \
+    --unattended \
+    --disableupdate \
+    --replace \
+    --labels z15 \
+    --ephemeral
+
+# Run one job.
+./run.sh
```
[the remark this answers is not in the document]
Comment (Dead2) [reply arch/s390/self-hosted-builder/actions-runner:58]: This error handling is not needed in this case

File: arch/s390/self-hosted-builder/actions-runner [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+fi
+if [[ $STATUS -ne 0 || $(date +%s) -ge $(date -d "$EXPIRES" +%s) ]]; then
+    # Refresh the cached registration token.
+    curl \
+        -X POST \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer $PAT_TOKEN" \
+        "https://api.github.com/repos/$REPO/actions/runners/registration-token" \
+        -o "$TOKEN_FILE"
+fi
```
[the remark this answers is not in the document]
Comment (Dead2) [reply arch/s390/self-hosted-builder/actions-runner:38]: The token file is stored privately in a docker, so no need for erasing it.
Timestamps are not so critical here that they need atomic checks.

**Pull request merged**